flang/lib/Lower/Intrinsics.cpp

```diff
@@ -625,13 +661,63 @@ IntrinsicLibrary::Implementation::genRuntimeCall(Context &context,
     }
   } else {
     // could not find runtime function
+    llvm::errs() << "missing intrinsic: " << context.name << "\n";
     assert(false && "no runtime found for this intrinsics");
     // TODO: better error handling ?
     //  - Try to have compile time check of runtime compltness ?
   }
   return {}; // gets rid of warnings
 }
 
+mlir::Value
+IntrinsicLibrary::Implementation::genConversion(Context &genCtxt,
+                                                MathRuntimeLibrary &) {
+  // There can be an optional kind in second argument.
+  assert(genCtxt.arguments.size() >= 1);
+  return genCtxt.builder->create<fir::ConvertOp>(
+      genCtxt.loc, genCtxt.getResultType(), genCtxt.arguments[0]);
+}
+
+// ABS
+mlir::Value
+IntrinsicLibrary::Implementation::genAbs(Context &genCtxt,
+                                         MathRuntimeLibrary &runtime) {
+  assert(genCtxt.arguments.size() == 1);
+  auto arg = genCtxt.arguments[0];
+  auto type = arg.getType();
+  if (type.isa<mlir::FloatType>() || type.isa<fir::RealType>()) {
```

Comment:
I think we have some generic type test floating around now...

Reply:
I agree we should, I could not find one (we have one or two local implementations). I created a isa_real and isa_complex (replacing FIROpsBuilder::isComplex) inside FIRType.h.

Reply:
That's a nice improvement. Moves more FIR type related stuff to the same place.